test/tla/Rec2.tla

```diff
@@ -18,7 +18,11 @@ Card(S) ==
   IF S = IntSet({})
   THEN 0
   ELSE
-    LET T == S \ {CHOOSE y \in S: TRUE} IN
+    \* CHOOSE is introduced with a LET definition, to fix its value, whatever it may be
+    \* Note that CHOOSE in APALACHE is non-deterministic and therefore
```

Comment:
This is one of the (two or three?) common patterns of using CHOOSE. Basically, we iterate over a set by using a fixed but unknown iteration order. Since we remove one element per iteration, it is not essential that CHOOSE is deterministic here.

Reply:
But since Apalache implements a non-deterministic version of CHOOSE (for efficiency), we have to call `CHOOSE` once in this code. Otherwise, Apalache produces a false-positive by picking two different elements from the same set `S`.

Reply:
Thanks for explaining!